jupyter-lib/api/src/main/kotlin/org/jetbrains/kotlinx/jupyter/api/Notebook.kt

```diff
@@ -11,6 +9,19 @@ interface Notebook {
      */
     val cellsList: Collection<CodeCell>
 
+    /**
+     * Current state of visible variables
+     */
+    val variablesState: Map<String, VariableState>
+
+    /**
+     * Stores info about useful variables in a cell.
+     * Key: cellId;
+     * Value: set of variable names.
+     * Useful <==> declarations + modifying references
+     */
+    var cellVariables: Map<Int, Set<String>>
```

Comment:
Do we still need it to be a `var`?

Reply:
We might create another field and make val in api. I think it might make sense since we do not want to allow change notebook.varsState from notebook

Reply:
Let's do so then